cmpctircd/SocketConnectorFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿using cmpctircd.Configuration;
+using System;
+
+namespace cmpctircd {
+    public class SocketConnectorFactory : ISocketConnectorFactory {
+        private readonly Log log;
+
+        public SocketConnectorFactory(Log log) {
+            this.log = log ?? throw new ArgumentNullException(nameof(log));
+        }
+
+        public SocketConnector CreateSocketConnector(IRCd ircd, ServerElement config) {
+            return new SocketConnector(log, ircd, config);
```

Comment:
This (and SocketListenerFactory) are only really just returning an object conditionless at the moment, I am guessing this is being done with a bigger picture in mind? I can see it nicely abstracts the Log creation but I don't really have the bigger picture in my head at the moment